### ISSUE TITLE
Audit: write up where we need to add docstring runnable examples

### ISSUE THREAD

**Issue by kmax12** [alteryx/evalml#207]
We should add examples to the doc strings of as many functions as possible. 

We can also set these up as doctests, so that when we build the docs, the output of the method call is tested against the example.

We do this in featuretools. For example: 
https://docs.featuretools.com/en/stable/generated/featuretools.primitives.Count.html#featuretools.primitives.Count

**Comment by chukarsten** [comment]
I think this is interesting, but I'm converting it into a spike because there's no clear left and right limits on what will close this issue.  I think if we have it generate an epic, that might be good, though.

**Comment by dsherry** [comment]
Yep, let's have the outcome of this issue be a list of the highest-importance places we should have docstring examples. IMO the automl search and pipeline APIs are at the top of that list. Things like AutoMLAlgorithm are lower priority because they're internal.

Then once we have that list we can file issues for specific parts, as good first issues, and pick em off.
